onnxscript/ir/_external_data.py

```diff
@@ -51,3 +75,209 @@ def set_base_dir(graph: _core.Graph | _core.GraphView, base_dir: str | os.PathLi
+            current_offset = tensor_info.offset
+            assert tensor is not None
+            raw_data = tensor.tobytes()
+            if current_offset is not None:
+                # Pad file to required offset if needed
+                file_size = data_file.tell()
+                if current_offset > file_size:
+                    data_file.write(b"\0" * (current_offset - file_size))
+            data_file.write(raw_data)
+
+
+def _convert_as_external_tensors(
+    external_data_info: list[tuple[_protocols.TensorProtocol, ExternalDataInfo]],
+    file_path: str | os.PathLike,
+) -> list[_core.ExternalTensor]:
+    """
+    Convert the tensors (stored within the values) written as external data to _core.ExternalTensor types.
+
+    Args:
+        external_data_info: A collection of external data information stored for each tensor to be written as external data.
+        file_path: Location to which external data is to be stored.
+    """
+    external_tensors: list[_core.ExternalTensor] = []
+    for tensor, tensor_info in external_data_info:
+        assert tensor is not None
+        external_tensor = _core.ExternalTensor(
+            file_path,
+            tensor_info.offset,
+            tensor_info.length,
+            tensor.dtype,  # type: ignore[arg-type]
+            shape=tensor.shape,  # type: ignore[arg-type]
+            name=tensor.name,  # type: ignore[arg-type]
+        )
+        external_tensors.append(external_tensor)
+    return external_tensors
+
+
+def convert_tensors_to_external(
+    tensors: list[_protocols.TensorProtocol],
+    abs_path: str | os.PathLike,
+) -> list[_core.ExternalTensor]:
+    """
+    This method takes in tensors to be converted to external tensors and returns the converted external tensors.
+
+    Args:
+        tensors: Tensors to be converted to external tensors
+        abs_path: Absolute path to which external data is to be stored.
+    """
+    external_data_info: list[tuple[_protocols.TensorProtocol, ExternalDataInfo]] = []
+    # Sort all tensors based on tensor sizes, in order to avoid unneccesarry alignment.
+    # All the smaller tensors are written earlier and alignment is performed for the larger tensors.
+    sorted_indices = sorted(range(len(tensors)), key=lambda i: tensors[i].nbytes)
+    tensors = [tensors[i] for i in sorted_indices]
+
+    current_offset = 0
+    for tensor in tensors:
+        tensor_info = _set_external_data(tensor, current_offset)
+        external_data_info.append((tensor, tensor_info))
+        current_offset = tensor_info.offset + tensor_info.length
+    _save_external_data(external_data_info, abs_path)
+
+    # Convert initializers to ExternalTensors
+    external_tensors = _convert_as_external_tensors(external_data_info, abs_path)
+    # Sort external_tensors based on original key order
+    external_tensors = [
+        external_tensors[i]
+        for i in sorted(range(len(external_tensors)), key=lambda i: sorted_indices[i])
```

Comment:
```suggestion
        for i in sorted_indices
```
maybe this?

Reply:
Also consider the original:

if sorted_indices is `[2,0,1]`

for sorting `[0,1,2]`, we get keys 0->2, 1->0, 2->1, so the sorted array is `[1,2,0]`. Is this expected?

Reply:
This might be incorrect I think, 
Let's say originally, its `[a, b, c]` with the order being b>a>c
Sorted tensors are now `[c, a, b]` and sorted indices are `[2, 0, 1]

With the original logic, 
keys are `[1, 2, 0]`, so we get back `[a, b, c]`

With the proposed logic, 
keys are `[2, 0, 1]`, so we get back `[c, a, b]`

Reply:
Thanks - I realized I was wrong with your explanation.